modules/validations/outdated_tags.js

```diff
@@ -17,58 +16,126 @@ let _nsi;
+        qids:          new Map(),              // Map wd/wp tag values -> qids
+        ids:           new Map()               // Map id -> NSI item
       };
 
-      // initialize name-suggestion-index matcher
-      _nsi.matcher.buildMatchIndex(d.brands);
+      _nsi.matcher = Matcher();
+      _nsi.matcher.buildMatchIndex(_nsi.data);
+      _nsi.matcher.buildLocationIndex(_nsi.data, locationManager.loco());
+
+      Object.keys(_nsi.data).forEach(tkv => {
+        const parts = tkv.split('/', 3);     // tkv = "tree/key/value"
+        const t = parts[0];
+        const k = parts[1];
+
+        // Collect primary keys  (e.g. "amenity", "craft", "shop", "man_made", "route", etc)
+        _nsi.keys.add(k);
+
+        const tree = _nsi.trees[t];     // e.g. "brands", "operators"
+        const mainTag = tree.mainTag;   // e.g. "brand:wikidata", "operator:wikidata", etc
 
-      // index all known wikipedia and wikidata tags
-      Object.keys(d.brands).forEach(kvnd => {
-        const brand = d.brands[kvnd];
-        const wd = brand.tags['brand:wikidata'];
-        const wp = brand.tags['brand:wikipedia'];
-        if (wd) { _nsi.wikidata[wd] = kvnd; }
-        if (wp) { _nsi.wikipedia[wp] = kvnd; }
+        const items = _nsi.data[tkv] || [];
+        items.forEach(item => {
+          // Cache NSI ids and main tags
+          item.mainTag = mainTag;
+          _nsi.ids.set(item.id, item);
+
+          // Cache Wikidata/Wikipedia values, for #6416
+          const wd = item.tags[mainTag];
+          const wp = item.tags[mainTag.replace('wikidata', 'wikipedia')];
+          if (wd)         _nsi.qids.set(wd, wd);
+          if (wp && wd)   _nsi.qids.set(wp, wd);
+        });
       });
 
+      _nsi.keys.add('building');  // fallback can match building=* for some categories
+
       return _nsi;
     })
     .catch(() => { /* ignore */ });
 
 
+  // Returns true if this tag key is a "namelike" tag that the NSI matcher would have indexed..
+  function isNamelike(k) {
+    const namePatterns = [
+      /^(flag:)?name$/i,                                             // e.g. `name`, `flag:name`
+      /^(brand|country|flag|operator|network|subject)$/i,
+      /^\w+_name$/i,                                                 // e.g. `alt_name`, `short_name`
+      /^(name|brand|country|flag|operator|network|subject):\w+$/i,   // e.g. `name:en`, `name:ru`
+      /^\w+_name:\w+$/i                                              // e.g. `alt_name:en`, `short_name:ru`
+    ];
+
+    return namePatterns.some(pattern => {
+      if (!pattern.test(k)) return false;    // k is not a name tag, skip
+
+      // There are a few exceptions to the namelike regexes.
+      // Usually a tag suffix contains a language code like `name:en`, `name:ru`
+      // but we want to exclude things like `operator:type`, `name:etymology`, etc..
+      if (/:(colour|type|left|right|etymology|wikipedia)$/.test(k)) return false;
```

Comment:
`name:pronunciation` is another common subkey of `name` that isn’t a language code. That said, IPA is very unlikely to match a name in NSI anyways.

Reply:
Yeah I _think_ that's ok - This code is just collecting values to attempt to match, so if collects a few extras it's probably fine, but will waste a few microseconds.

Reply:
I removed it anyway in a32553595